### PR TITLE
gh-122470: Improve REPL welcome message by indicating build type

### PR DESCRIPTION
**Pull request by picnixz**
I don't know whether there are other optimizations that I don't know about but I think this should be sufficient. Also, I don't know if I need to do something more for the C API and whether the news should be in C API or the Build section (I'll write a WhatsNew tomorrow and fix errors tomorrow as well).

* Issue: gh-122470

----
📚 Documentation preview 📚: https://cpython-previews--122473.org.readthedocs.build/

### CODE REVIEW

File: Doc/c-api/init.rst [hunk 1]
```diff
@@ -669,6 +670,22 @@ Process-wide parameters
    ``sys.version``.
 
 
+.. c:function:: const char* Py_GetConfigInfo()
+
+   Return information about how the ``configure`` script was invoked, e.g.::
+
+      "release"
```
Comment (picnixz) [inline Doc/c-api/init.rst:677]: I'll improve that (it's not really clear)

**Review by picnixz — COMMENTED** [review]
TODO for me

**Comment by ned-deily** [comment]
I've [commented in more detail](https://github.com/python/cpython/issues/122470#issuecomment-2258984205) on the issue but, to summarize here, I question whether adding a change that affects so many files and parts of Python is worth the maintenance cost considering the very small audience for such a change and that there are existing mechanisms to access this information.

File: Misc/stable_abi.toml [hunk 2]
```diff
@@ -1628,6 +1628,8 @@
     added = '3.2'
 [function.Py_GetCompiler]
     added = '3.2'
+[function.Py_GetConfigInfo]
+    added = '3.14'
```
Comment (encukou) [inline Misc/stable_abi.toml:1632]: The function itself is not useful without also exposing the `PY_CONFIG*` constants. And those are too CPython-specific for limited API.

**Review by encukou — CHANGES_REQUESTED** [review]

File: Misc/stable_abi.toml (hunk 2 above)
Comment (picnixz) [reply Misc/stable_abi.toml:1632]: Actually the other functions used for constructing the version string were also exposed, so I assumed that mine should have been as well ... But honestly I didn't want to :/

As for the constants I think I forgot about them (first time I need to do that kind of thing...)

**Comment by picnixz** [comment]
Closing as not planned (there are simpler alternatives).

**Pull request closed without merging**